Frontend/student-hub/src/pages/Friends.js

```diff
@@ -0,0 +1,279 @@
+        person.major.toLowerCase().includes(searchTerm)
+        );
+
+    const filteredRecommendedFriends = recommendedFriends.filter(friend =>
+        friend.name.toLowerCase().includes(searchTerm)  ||
+        friend.interest.toLowerCase().includes(searchTerm) ||
+        friend.school.toLowerCase().includes(searchTerm) ||
+        friend.major.toLowerCase().includes(searchTerm)
+        );
+
+    const filteredAvailableFriend = availableFriend.filter(person =>
+        person.name.toLowerCase().includes(searchTerm) ||
+        person.interest.toLowerCase().includes(searchTerm) ||
+        person.school.toLowerCase().includes(searchTerm) ||
+        person.major.toLowerCase().includes(searchTerm)
+        );
+
+    //update filtered data
+    setFilteredData({
+        friends: filteredFriends,
+        recommendedFriends: filteredRecommendedFriends,
+        availableFriend: filteredAvailableFriend
+    });
+};
+
+
+//rendering
+return (
+    <Container maxwidth="sm">
+        <Grid container spacing={2} alignItems="center" justify="center" style={{ minHeight: '80vh'}}>
+            {/*Search bar */}
+            <Grid item xs={12}>
+                <SearchContainer>
+                    <SearchInput
+                    placeholder="Search Friends"
+                    inputProps={{ 'aria-label': 'search Friends'}}
+                    value={searchTerm}
+                    onChange={handleSearch}
+                    />
+                    <IconButton aria-label="search">
+                        <SearchIcon />
+                    </IconButton>
+                </SearchContainer>
+            </Grid>
+            {/* Friends List */}
+            <Grid item xs={12}>
+                <Typography variant="h5">Friends</Typography>
+                <List>
+                    {filteredData.friends.map((person) => (
+                        <ListItem key={person.id}>
+                            <ListItemText primary={person.name} secondary={`${person.interest}, ${person.school}, ${person.major}`} />
+                            <ListItemSecondaryAction>
+                            <IconButton edge="end" aria-label="remove" onClick={() => handleRemoveFriend(person.id)}>
+                                <Remove />
+                            </IconButton>
+                            </ListItemSecondaryAction>
+                        </ListItem>
+                    ))}
+                </List>
+            </Grid>
+            {/* Recommended Friends */}
+            <Grid item xs={12}>
+                <Typography variant="h6">Recommended Friends</Typography>
+                <List>
+                    {filteredData.recommendedFriends.map((friend) => (
+                        <ListItem key={friend.id}>
+                            <ListItemText primary={friend.name || 'Name not available'} secondary={`${friend.interest || 'Interest not found'}, ${friend.school || 'School not found'}, ${friend.major || 'Major not found'}`} />
```

Comment:
UI nit: These null states are a lot of text, can't we just leave this empty? We want to generally keep the information lightweight rather than adding too much and making the user think/read a lot

```suggestion
                            <ListItemText primary={friend.name} secondary={`${friend.interest}, ${friend.school}, ${friend.major}`} />
```